### ISSUE TITLE
ML Training Pipeline: Wire batch sim decision logging to training module

### ISSUE THREAD

**Issue by KoalaTrapLord** [KoalaTrapLord/commander-ai-lab#63]
## Problem

The ML training pipeline (`routes/ml.py`) depends on **`ml-decisions-*.jsonl`** files for training data, but batch simulations (`routes/lab.py`) only produce these files when ML logging is explicitly enabled. By default, `_ml_logging_enabled = False`, meaning all batch runs generate `batch-{id}.json` result summaries but **no decision-level training data**.

This creates a silent gap: users can run hundreds of sims and then find zero training data when they trigger `POST /api/ml/train`.

## Current Data Flow

```
Batch Sim ──> batch-{id}.json        (always written — win rates, metadata)
Batch Sim ──> ml-decisions-{id}.jsonl (ONLY if _ml_logging_enabled == True)
                    │
                    ▼
         ml.data.dataset_builder.build_dataset()
                    │
                    ▼
         train.npz / val.npz / test.npz
                    │
                    ▼
         POST /api/ml/train  (supervised)
         POST /api/ml/train/ppo  (reinforcement)
```

## Files Involved

| File | Role |
|------|------|
| `routes/shared.py` | Defines `_ml_logging_enabled` global toggle (default `False`) |
| `routes/lab.py` | `run_batch_subprocess()` — passes logging flag to Forge subprocess |
| `routes/ml.py` | `POST /api/ml/toggle` — sets the flag; `POST /api/ml/train` — consumes JSONL data |
| `ml/data/dataset_builder.py` | `build_dataset(results_dir)` — reads `ml-decisions-*.jsonl` files |
| `services/forge_runner.py` | Forge JAR invocation — must emit decision snapshots when flag is on |

### Phase 1: Verify Current State
- [ ] Confirm `_ml_logging_enabled` default in `routes/shared.py` — currently `False`
- [ ] Check if `run_batch_subprocess` in `routes/shared.py` actually reads `_ml_logging_enabled` and passes it to the Forge JAR command
- [ ] Verify the Forge Java side writes `ml-decisions-{batchId}.jsonl` when the flag is passed
- [ ] Run `GET /api/ml/data/status` and confirm `decisionFiles` is empty (no JSONL files exist)
- [ ] Run `GET /api/ml/status` and confirm `total_decisions: 0`

### Phase 2: Enable Logging and Generate Training Data
- [ ] Call `POST /api/ml/toggle?enable=true` to flip the flag
- [ ] Run a batch simulation via `POST /api/lab/start` with 2+ decks and 50+ games
- [ ] After completion, verify `results/ml-decisions-{batchId}.jsonl` was created alongside `results/batch-{batchId}.json`
- [ ] Inspect the JSONL file — each line should be a valid JSON decision snapshot with game state features
- [ ] Run `GET /api/ml/data/status` again and confirm the new file appears with a non-zero decision count
- [ ] Run `GET /api/ml/decisions/{filename}?limit=5` to spot-check snapshot structure

### Phase 3: Build Dataset and Train
- [ ] Call `POST /api/ml/train` with `{"rebuildDataset": true, "epochs": 50}`
- [ ] Monitor via `GET /api/ml/train/status` — confirm phases: `building` -> `training` -> `evaluating` -> `done`
- [ ] Verify `ml/models/train.npz`, `val.npz`, `test.npz` are created with correct shapes
- [ ] Verify `ml/models/checkpoints/best_policy.pt` is written
- [ ] Check `ml/models/checkpoints/eval_results.json` for test set accuracy
- [ ] Confirm policy auto-reloads: `GET /api/ml/model` should show `loaded: true`

### Phase 4: Validate End-to-End Inference
- [ ] Call `POST /api/ml/predict` with a sample decision snapshot — should return a valid action
- [ ] Call `POST /api/ml/predict/batch` with multiple snapshots
- [ ] Run a new batch sim with `useLearnedPolicy: true` to confirm the trained policy is used in-game

### Phase 5: PPO Reinforcement Training (Optional)
- [ ] Call `POST /api/ml/train/ppo` with `{"loadSupervised": "best_policy.pt", "iterations": 100}`
- [ ] Monitor via `GET /api/ml/train/ppo/status`
- [ ] After completion, verify `ml/models/checkpoints/best_ppo.pt` exists
- [ ] Run `POST /api/ml/tournament` to evaluate supervised vs PPO vs heuristic vs random

### Phase 6: Improvement Considerations
- [ ] Consider defaulting `_ml_logging_enabled = True` so training data accumulates automatically
- [ ] Add a startup warning in `lab_api.py` if no JSONL files exist in `results/`
- [ ] Add a UI indicator on the Sim Lab page showing whether ML logging is active
- [ ] Consider auto-enabling ML logging when a user navigates to the Training tab
- [ ] Add data retention policy — JSONL files can grow large over many batch runs

## Acceptance Criteria

1. Running a batch sim with ML logging enabled produces a valid `ml-decisions-{id}.jsonl` file
2. `POST /api/ml/train` successfully builds a dataset from those files and trains a policy model
3. The trained model loads and returns predictions via `/api/ml/predict`
4. A sim run with `useLearnedPolicy: true` uses the trained model for decisions

**Comment by KoalaTrapLord** [comment]
Closing this issue — the root cause has been resolved.

`_ml_logging_enabled` in `routes/shared.py` now defaults to `True`, so batch sims produce `ml-decisions-*.jsonl` files by default. The silent data gap described in this issue no longer exists.

Additionally, the closed-loop ML pipeline has been fully wired:
- `DecisionExporter` is integrated into `self_play.py` (PPO training loop)
- `dataset_builder.py` consumes the JSONL output
- Supervised and reinforcement training endpoints are functional

Phase 6 improvement considerations (UI indicator, data retention policy, etc.) can be tracked separately if needed.
